Algorithm/UniverseDefinitions.cs

```diff
@@ -23,35 +27,160 @@ namespace QuantConnect.Algorithm
+        }
 
         /// <summary>
-        /// Gets a helper that provides methods for creating constituent universes
+        /// Creates a universe for the constituents of the provided ETF <paramref name="symbol"/>
         /// </summary>
-        public ConstituentUniverseDefinitions Constituent { get; }
+        /// <param name="symbol">ETF Symbol to get constituents for</param>
+        /// <param name="universeSettings">Universe settings</param>
+        /// <param name="universeFilterFunc">Function to filter universe results</param>
+        /// <returns>New ETF constituents Universe</returns>
+        public Universe ETF(
+            Symbol symbol,
+            UniverseSettings universeSettings = null, 
+            Func<IEnumerable<ETFConstituentData>, IEnumerable<Symbol>> universeFilterFunc = null)
+        {
+            return new ETFConstituentsUniverse(symbol, universeSettings ?? _algorithm.UniverseSettings, universeFilterFunc);
+        }
+        
+        /// <summary>
+        /// Creates a universe for the constituents of the provided ETF <paramref name="symbol"/>
+        /// </summary>
+        /// <param name="symbol">ETF Symbol to get constituents for</param>
+        /// <param name="universeSettings">Universe settings</param>
+        /// <param name="universeFilterFunc">Function to filter universe results</param>
+        /// <returns>New ETF constituents Universe</returns>
+        public Universe ETF(
+            Symbol symbol,
+            UniverseSettings universeSettings = null, 
+            PyObject universeFilterFunc = null)
+        {
+            return ETF(symbol, universeSettings, universeFilterFunc.ConvertPythonUniverseFilterFunction<ETFConstituentData>());
+        }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="UniverseDefinitions"/> class
+        /// Creates a universe for the constituents of the provided <paramref name="indexTicker"/>
         /// </summary>
-        /// <param name="algorithm">The algorithm instance, used for obtaining the default <see cref="UniverseSettings"/></param>
-        public UniverseDefinitions(QCAlgorithm algorithm)
+        /// <param name="indexTicker">Ticker of the index to get constituents for</param>
+        /// <param name="market">Market of the index</param>
+        /// <param name="universeSettings">Universe settings</param>
+        /// <param name="universeFilterFunc">Function to filter universe results</param>
+        /// <returns>New index constituents Universe</returns>
+        public Universe Index(
+            string indexTicker,
+            string market = null,
+            UniverseSettings universeSettings = null,
+            Func<IEnumerable<ETFConstituentData>, IEnumerable<Symbol>> universeFilterFunc = null)
+        {
+            return Index(
+                Symbol.Create(indexTicker, SecurityType.Index, market), 
```

Comment:
The overloads accepting a string ticker throw a null reference exception, market can't be null. Let's add a tests covering the cases too